src/main/java/de/tum/in/www1/artemis/repository/LectureRepository.java

```diff
@@ -17,16 +17,31 @@
 @Repository
 public interface LectureRepository extends JpaRepository<Lecture, Long> {
 
-    @Query("select lecture from Lecture lecture left join fetch lecture.attachments WHERE lecture.course.id = :#{#courseId}")
-    Set<Lecture> findAllByCourseId(@Param("courseId") Long courseId);
+    @Query("""
+            SELECT lecture
+            FROM Lecture lecture
+            LEFT JOIN FETCH lecture.attachments
+            WHERE lecture.course.id = :#{#courseId}
+            """)
+    Set<Lecture> findAllByCourseIdWithAttachments(@Param("courseId") Long courseId);
+
+    @Query("""
+                        SELECT lecture
+                        FROM Lecture lecture
+                        LEFT JOIN FETCH lecture.attachments
+                        LEFT JOIN FETCH lecture.lectureUnits
+                        WHERE lecture.course.id = :#{#courseId}
```

Comment:
The indentation here looks off.

Reply:
Good Catch! 09c53aa3fe1d67dfc7e1c2a114156a8c71725b90